security-model.md

```diff
@@ -0,0 +1,105 @@
+# Security Model
+
+This security model aims to clarify the project's current security position against industry 
+recommendations and general security best practices. The information provided here should make 
+it easier for security teams from highly regulated sectors to assess and use SPO for their workloads.
+
+The SPO project strives to maintain the [least privilege principle], by trying to run with 
+the bare minimum privileges for the least amount of time possible. This approach decreases
+the attack surface and the window of opportunity in which it can be abused.
+
+Note that this is an on-going effort, and may change over time as new features are developed.
+
+## Host Isolation (a.k.a. Pod Security / Security Context)
+
+Details the security requirements to run SPO from a host isolation perspective. 
+
+### Initialisation
+
+Seccomp leverages an init container to setup an initial symlink as `root`, so it can then operate
+ as a non-root user.
+
+During initialisation it also uses the capabilities:
+
+- `CHOWN`
+- `FOWNER`
+- `FSETID`
+- `DAC_OVERRIDE`
+
+
+### Running Mode
+
+The running permissions for the three core technologies supported: 
+
+|                                  | Seccomp | SELinux | AppArmor |
+|----------------------------------|---------|---------|----------|
+|               Requires root user |    No   |   Yes   |    Yes   |
+|                 Requires HostPID |    No   |    ?    |    Yes   |
+|  Requires "privileged container" |    No   |   Yes   |    Yes   |
+|     Requires SSH Access to nodes |    No   |    No   |    No    |
+| Access to host's mount namespace |    No   |    ?    | On Demand|
```

Comment:
Sure, going through it again it does not come across as clear as I intented it to be.

On a follow-up PR I think this can be replaced by breaking down the "host paths" for each main feature (SAS) and the access (rw) they require. So regardless of this is implemented, for Seccomp to work it requires `rw` access to `{kubelet_path}/seccomp`. And the same for selinux/apparmor.

Reply:
Right, selinuxd needs to write to the hosts's selinux db. Thanks for the explanation.